### ISSUE TITLE
Security concern in the HTTP headers Server document

### ISSUE THREAD

**Issue by lkc0626** [mdn/content#33543]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server#directives

### What specific section or headline is this issue about?

Directives

### What information was incorrect, unhelpful, or incomplete?

The HTTP headers Server documentation includes potentially misleading security advice. 
It mentions that having "Server" information in the HTTP header can expose the server to exploitation by attackers. 
However, the directive section suggests that revealing Apache versions helps browsers work around bugs. 
Instead, developers should patch bugs without exposing vulnerable information to potential attackers. 
Thus, revealing server information contradicts the security warnings in the document.

Below is the statement from the document:
"How much detail to include is an interesting balance to strike; exposing the OS version is probably a bad idea, as mentioned in the earlier warning about overly-detailed values. However, exposed Apache versions helped browsers to work around a bug of the versions with [Content-Encoding](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) and [Range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range) in combination."

### What did you expect to see?

Update the document so that contradicting statements will be removed.

### Do you have any supporting links, references, or citations?

RFC-2616 states that server information should be confidential.
https://datatracker.ietf.org/doc/html/rfc2616#section-15.1.1

**Comment by BrianJDrake** [comment]
Despite my comment above, this issue does raise an important point. Including the `Server` directive to help clients work around server bugs is definitely a bad idea:

1. It may be a security concern, even if the MDN page as currently written does a poor job of explaining that.
1. It is not the clients' job to work around server bugs.

On the other hand, omitting the `Server` directive as a security measure may also be a bad idea (even if it is consistent with the RFC), as it may lead to a false sense of security.

Here are some more general comments:

1. The MDN `Server` page references the `User-Agent` page for the format, but we could also reference that page for security advice (or lack of). Why are we so worried about servers telling clients what software they use, but not at all worried about clients telling servers what software they use?
1. If you are relying on this sort of security advice, you are probably not qualified to operate a server in the first place. MDN should carefully consider what sort of security advice it includes, and link to other security resources whenever it does include this sort of security advice.

**Comment by bsmth** [comment]
I agree with the opinion that obscuring the server version is not very helpful, although I can see why it might be tempting to do so if you're stuck running old software that you cannot patch: https://en.wikipedia.org/wiki/Security_through_obscurity

We should revise or remove this advice entirely.
